pkg/authorization/toplevel_org_authorizer.go

```diff
@@ -95,20 +111,42 @@ func (a *topLevelOrgAccessAuthorizer) Authorize(ctx context.Context, attr author
 		if isAuthenticated && (isUser || isServiceAccountFromRootCluster) {
 			return a.delegate.Authorize(ctx, attr)
 		}
+		kaudit.AddAuditAnnotations(
+			ctx,
+			TopLevelContentAuditDecision, DecisionNoOpinion,
+			TopLevelContentAuditReason, "only authenticated users and serviceaccounts in root may access root",
+		)
 		return authorizer.DecisionNoOpinion, WorkspaceAcccessNotPermittedReason, nil
 	}
 
 	// get org in the root
 	requestTopLevelOrgName, ok := topLevelOrg(cluster.Name)
 	if !ok {
+		kaudit.AddAuditAnnotations(
+			ctx,
+			TopLevelContentAuditDecision, DecisionNoOpinion,
+			TopLevelContentAuditReason, "not part of root workspace hierarchy",
```

Comment:
let's add the offending cluster name for debugability: `TopLevelContentAuditReason, fmt.Sprintf("access not permitted, workspace %q is not part of root workspace hierarchy", cluster.Name)`

Reply:
Isn't it already there with the `tenancy.kcp.dev/workspace` annotation?

Reply:
if so, then perfect 👍